code/network/multi_sexp.cpp

```diff
@@ -819,6 +819,6 @@ bool multi_get_ushort(ushort &value)
 void multi_discard_remaining_callback_data()
 {
 	if (!multi_sexp_discard_operator()) {
-		Warning(LOCATION, "Attempt to discard remaining data failed! Callback for operator %d lacks proper termination. Entire packet may be corrupt. Discarding remaining packet"); 
+		Warning(LOCATION, "Attempt to discard remaining data failed! Callback for operator lacks proper termination. Entire packet may be corrupt. Discarding remaining packet");
```

Comment:
I don't think it's that simple though, as plumbing through the operator will be a big pain. Note we can't directly call `multi_sexp_get_next_operator()` here as we do in `multi_sexp_eval()`.

Also, the only paths to this function are via `multi_sexp_eval()`, which does directly call `multi_sexp_discard_operator()` and report errors if an invalid operator is sent.

Happy to review suggested patches though from others who know the SEXP code better. Or it could be addressed in a follow-up PR.